core/src/main/scala/org/apache/spark/shuffle/IndexShuffleBlockResolver.scala

```diff
@@ -202,20 +210,21 @@ private[spark] class IndexShuffleBlockResolver(
     // class of issue from re-occurring in the future which is why they are left here even though
     // SPARK-22982 is fixed.
     val channel = Files.newByteChannel(indexFile.toPath)
-    channel.position(blockId.reduceId * 8L)
+    channel.position(startReduceId * 8L)
     val in = new DataInputStream(Channels.newInputStream(channel))
     try {
       val offset = in.readLong()
+      channel.position(endReduceId * 8L)
       val nextOffset = in.readLong()
```

Comment:
nit: we probably should rename "offset", "nextOffset" to "startOffset", "endOffset"

Reply:
Thanks, done in afd74e5.